vllm/distributed/communication_op.py

```diff
@@ -315,3 +322,35 @@ def broadcast_tensor_dict(
         for async_handle in async_handles:
             async_handle.wait()
     return tensor_dict
+
+
+def send_next_rank(tensors: List[torch.Tensor]) -> None:
+    """Send the tensors to the next pipeline model parallel rank."""
+    pynccl_comm = get_pp_pynccl_communicator()
+    if (pynccl_comm is not None and not pynccl_comm.disabled):
+        for tensor in tensors:
+            pynccl_comm.send(tensor)
+    else:
+        for tensor in tensors:
+            torch.distributed.send(tensor,
+                                   get_pipeline_model_parallel_next_rank(),
+                                   get_pipeline_model_parallel_group())
+
+
+def recv_prev_rank(num_tensors: int, size: torch.Size, dtype: torch.dtype,
+                   device: torch.device) -> List[torch.Tensor]:
+    """Receive tensors from the previous pipeline model parallel rank."""
```

Comment:
Can you add docstring for inputs and output?

Reply:
Done